qutip/core/metrics.py

```diff
@@ -105,12 +108,104 @@ def fidelity(A, B):
     return float(np.real(np.sqrt(eig_vals[eig_vals > 0]).sum()))
 
 
-def process_fidelity(U1, U2, normalize=True):
+def _process_fidelity_to_id(oper):
+    """
+    Internal function returning the process fidelity of a quantum channel
+    to the identity quantum channel.
+    Parameters
+    ----------
+    oper : :class:`qutip.Qobj`/list
+        A unitary operator, or a superoperator in supermatrix, Choi or
+        chi-matrix form, or a list of Kraus operators
+    Returns
+    -------
+    fid : float
     """
-    Calculate the process fidelity given two process operators.
+    msg = 'The process fidelity to identity is only defined for ' \
+          'dimension preserving channels.'
+    if isinstance(oper, list):  # oper is a list of Kraus operators
+        d = oper[0].shape[0]
+        if oper[0].shape[1] != d:
+            raise TypeError(msg)
+        return np.sum([np.abs(k.tr()) ** 2 for k in oper]) / d ** 2
+    elif oper.type == 'oper':  # interpret as unitary
+        d = oper.shape[0]
+        if oper.shape[1] != d:
+            raise TypeError(msg)
+        return np.abs(oper.tr()) ** 2 / d ** 2
+    elif oper.type == 'super':
+        d = np.prod(oper.dims[0][0])
+        if np.prod(oper.dims[1][0]) != d:
+            raise TypeError(msg)
+        if oper.superrep == 'chi':
+            return oper[0, 0].real / d ** 2
+        else:  # oper.superrep is either 'super' or 'choi':
+            return to_super(oper).tr().real / d ** 2
+
+
+def _kraus_or_qobj_to_choi(oper):
+    if isinstance(oper, list):
+        return kraus_to_choi(oper)
+    else:
+        return to_choi(oper)
+
+
+def process_fidelity(oper, target=None):
     """
-    out = (U1 * U2).tr()
-    return out / (U1.tr() * U2.tr()) if normalize else out
+    Returns the process fidelity of a quantum channel to the target
+    channel, or to the identity channel if no target is given.
+    The process fidelity between two channels is defined as the state
+    fidelity between their normalized Choi matrices.
+    Parameters
```

Comment:
```suggestion

    Parameters
```
Just adding some paragraph spacing.